kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/predicates/In.java

```diff
@@ -116,8 +116,11 @@ public String toString() {
   }
 
   public String toString(Schema schema) {
-    return Joiner.on(',').join(
-        Iterables.transform(set, new ToString<T>(schema)));
+    // we want the string this returns to be consistent
+    // ensure the set is sorted before calling toString on it.
+
+    Iterable<String> strings = Iterables.transform(set, new ToString<T>(schema));
+    return Joiner.on(',').join(Sets.newTreeSet(strings));
```

Comment:
I'm not sure I'm clear on what happens here. Generally, we are using a `LinkedHashMap` or `LinkedHashSet` to preserve the order. If we aren't doing that somewhere, then I think it is a bug and should be fixed.

URIs should only be normalized when needed, to avoid annoyance. I think that's what you do in this case, but I wanted to make sure.